core/src/test/java/io/questdb/griffin/engine/groupby/TimestampSamplerFactoryTest.java

```diff
@@ -106,6 +106,31 @@ public void testSeconds() throws NumericException, SqlException {
         }
     }
 
+    @Test
+    public void testMicros() throws NumericException, SqlException {
+        StringSink sink = new StringSink();
+        for (int k = 0; k < 61; k++) {
```

Comment:
micros `k < 1000001`, and then the loop would take forever. May I suggest to use something like:

```
    Rnd rand = new Rnd();
    for (int round = 0; round < 1000; round++) {
         int k = rand.nextInt(1000001);
         ...
``` 

(rounds < sqrt(micros))

Reply:
Thanks! And I think it is possible to add a `testMillis()` method, and both `k < 1001`, right?